commit 59482b5a1cf1f78854ae5f88fd9e68ecedf793d6
Author: Keiichi Fujino <kfujino@apache.org>
Date:   Mon Jun 3 07:24:09 2013 +0000

    Add maxWait attribute that the senderPool will wait when there are no available senders.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1488871 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/tribes/transport/PooledSender.java b/java/org/apache/catalina/tribes/transport/PooledSender.java
index 573d59b..5abeaf2 100644
--- a/java/org/apache/catalina/tribes/transport/PooledSender.java
+++ b/java/org/apache/catalina/tribes/transport/PooledSender.java
@@ -42,6 +42,7 @@ public abstract class PooledSender extends AbstractSender implements MultiPointS
 
     private final SenderQueue queue;
     private int poolSize = 25;
+    private long maxWait = 3000;
     public PooledSender() {
         queue = new SenderQueue(this,poolSize);
     }
@@ -49,7 +50,7 @@ public abstract class PooledSender extends AbstractSender implements MultiPointS
     public abstract DataSender getNewDataSender();
 
     public DataSender getSender() {
-        return queue.getSender(getTimeout());
+        return queue.getSender(getMaxWait());
     }
 
     public void returnSender(DataSender sender) {
@@ -89,6 +90,14 @@ public abstract class PooledSender extends AbstractSender implements MultiPointS
         return poolSize;
     }
 
+    public long getMaxWait() {
+        return maxWait;
+    }
+
+    public void setMaxWait(long maxWait) {
+        this.maxWait = maxWait;
+    }
+
     @Override
     public boolean keepalive() {
         //do nothing, the pool checks on every return
diff --git a/webapps/docs/config/cluster-sender.xml b/webapps/docs/config/cluster-sender.xml
index f5461bb..bdb4b4d 100644
--- a/webapps/docs/config/cluster-sender.xml
+++ b/webapps/docs/config/cluster-sender.xml
@@ -162,14 +162,18 @@
       </attribute>
     </attributes>
   </subsection>
-  <subsection name="PooledParallelSender Attributes">
+  <subsection name="Common PooledSender Attributes">
     <attributes>
       <attribute name="poolSize" required="false">
         The maximum number of concurrent connections from A to B.
         The value is based on a per-destination count.
         The default value is <code>25</code>
       </attribute>
-
+      <attribute name="maxWait" required="false">
+        The maximum number of milliseconds that the senderPool will wait when
+        there are no available senders. The default value is <code>3000</code>
+        milliseconds.(3 seconds).
+      </attribute>
     </attributes>
   </subsection>
 </section>
